crates/scmi/src/devices/common.rs

```diff
@@ -0,0 +1,531 @@
+    /// Returns the inner [Sensor] instance, mutable.
+    fn sensor_mut(&mut self) -> &mut Sensor;
+
+    /// Performs any non-default initialization on the sensor.
+    ///
+    /// If the initialization fails, a corresponding error message is
+    /// returned.
+    fn initialize(&mut self) -> Result<(), String> {
+        Ok(())
+    }
+
+    /// Returns the id of the SCMI protocol used to communicate with the
+    /// sensor.
+    ///
+    /// Usually no need to redefine this.
+    fn protocol(&self) -> ProtocolId {
+        SENSOR_PROTOCOL_ID
+    }
+
+    /// Returns an error message about invalid property `name`.
+    ///
+    /// Usually no need to redefine this.
+    fn invalid_property(&self, name: &String) -> Result<(), String> {
+        Result::Err(format!("Invalid device option: {name}"))
+    }
+
+    /// Processes a device property specified on the command line.
+    ///
+    /// The function is called on all the device properties from the command line.
+    fn process_property(&mut self, name: &String, _value: &str) -> Result<(), String> {
+        self.invalid_property(name)
+    }
+
+    /// Returns the number of axes of the given sensor.
+    ///
+    /// If the sensor provides just a scalar value, 0 must be returned (the
+    /// default return value here).  Otherwise a non-zero value must be
+    /// returned, even for vector sensors with a single access.
+    fn number_of_axes(&self) -> u32 {
+        0
+    }
+
+    /// Formats the unit of the given `axis` for SCMI protocol.
+    ///
+    /// Usually no need to redefine this.
+    fn format_unit(&self, axis: u32) -> u32 {
+        (self.unit_exponent(axis) as u32 & 0x1F) << 11 | u32::from(self.unit())
+    }
+
+    /// Returns SCMI description of the sensor.
+    ///
+    /// Usually no need to redefine this.
+    fn description_get(&self) -> DeviceResult {
+        // Continuous update required by Linux SCMI IIO driver
+        let low = 1 << 30;
+        let n_axes = self.number_of_axes();
+        let high = if n_axes > 0 {
+            n_axes << 16 | 1 << 8
+        } else {
+            self.format_unit(0)
+        };
+        let name = self.sensor().name.clone();
+        let values: MessageValues = vec![
+            // attributes low
+            MessageValue::Unsigned(low),
+            // attributes high
+            MessageValue::Unsigned(high),
+            // name, up to 16 bytes with final NULL (non-extended version)
+            MessageValue::String(name, MAX_SIMPLE_STRING_LENGTH),
+        ];
+        Ok(values)
+    }
+
+    /// Returns the SCMI unit of the sensor.
+    fn unit(&self) -> u8 {
+        scmi::SENSOR_UNIT_UNSPECIFIED
+    }
+
+    /// Returns the decadic exponent to apply to the sensor values.
+    fn unit_exponent(&self, _axis: u32) -> i8 {
+        0
+    }
+
+    /// Returns the prefix of axes names.
+    ///
+    /// Usually no need to redefine this.
+    fn axis_name_prefix(&self) -> String {
+        String::from("axis")
+    }
```

Comment:
```suggestion
    fn axis_name_prefix(&self) -> &'static str {
        "axis"
    }
```

Reply:
In this particular implementation this would be OK. But implementations of the trait may redefine this and construct the axis name dynamically and then "&'static str" return type may not be the best idea.